### PR TITLE
Revocation Handle Comm

### LINKED ISSUE #21: add ability to commit to the revocation handle
Currently, we can generate an Idemix signature attaching a commitment to the enrolment ID (EID) with a proof that it is the same of that in the signature. This can be enabled by setting the `SignatureType` to `EidNym`.

This issue asks to add a new SignatureType that attaches both a commitment to EID and to Revocation Handle (RH).

### PR DESCRIPTION
**Pull request by adecaro**
This PR adds commitment functionality to the revocation handle attribute. This is achieved through the following changes:

- Introduction of new revocation handle types pertaining to signatures, signature types, and signature options
- More generalized attribute functionality to handle both enrollment id and revocation handle commitment schemes
- Addition of ability to generate a pseudonym of a revocation handle (commitment)
- Addition of ability to generate a signature containing both an enrollment id pseudonym and a revocation handle pseudonym
- Functionality for auditing and verifying the new signature type

PR motivation: https://github.com/IBM/idemix/issues/21

### CODE REVIEW

File: bccsp/schemes/dlog/bridge/signaturescheme.go [hunk 1]
```diff
@@ -142,6 +144,60 @@ func (s *SignatureScheme) AuditNymEid(
...
+		}
+		return sig.AuditNymRh(
+			iipk.PK,
+			rhAttr,
+			rhIndex,
+			RNymRh,
+			s.Idemix.Curve,
+			s.Translator,
+		)
+	case bccsp.AuditExpectEidNymRhNym:
```
Comment (ale-linux) [inline bccsp/schemes/dlog/bridge/signaturescheme.go:184]: untested

File: bccsp/schemes/dlog/bridge/signaturescheme.go [hunk 2]
```diff
@@ -125,6 +125,8 @@ func (s *SignatureScheme) AuditNymEid(
 			s.Idemix.Curve,
 			s.Translator,
 		)
+	case bccsp.AuditExpectEidNymRhNym:
```
Comment (ale-linux) [inline bccsp/schemes/dlog/bridge/signaturescheme.go:128]: untested

**Review by ale-linux — CHANGES_REQUESTED** [review]
Fantastic, thanks Angelo! Just 2 missing tests and we're good to merge!

File: bccsp/schemes/dlog/bridge/signaturescheme.go (hunk 1 above)
Comment (adecaro) [reply bccsp/schemes/dlog/bridge/signaturescheme.go:184]: test added

File: bccsp/schemes/dlog/bridge/signaturescheme.go (hunk 2 above)
Comment (adecaro) [reply bccsp/schemes/dlog/bridge/signaturescheme.go:128]: test added

**Comment by adecaro** [comment]
Hi @ale-linux , I have extended the test to cover the cases you pointed to. Please, have another look :)

**Review by ale-linux — APPROVED** [review]

**Pull request merged**